src/plugins/data/public/ui/dataset_navigator/dataset_navigator.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { EuiButtonEmpty, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import { SavedObjectsClientContract, SimpleSavedObject } from 'opensearch-dashboards/public';
+import { map, scan } from 'rxjs/operators';
+import { ISearchStart } from '../../search/types';
+import { IIndexPattern } from '../..';
+import { getUiService, getIndexPatterns, getSearchService } from '../../services';
+
+const getClusters = async (savedObjectsClient: SavedObjectsClientContract) => {
+  return await savedObjectsClient.find({
+    type: 'data-source',
+    perPage: 10000,
+  });
+};
+
+export const searchResponseToArray = (showAllIndices: boolean) => (response) => {
+  const { rawResponse } = response;
+  if (!rawResponse.aggregations) {
+    return [];
+  } else {
+    return rawResponse.aggregations.indices.buckets
+      .map((bucket: { key: string }) => {
+        return bucket.key;
+      })
+      .filter((indexName: string) => {
+        if (showAllIndices) {
+          return true;
+        } else {
+          return !indexName.startsWith('.');
```

Comment:
probably doesn't matter but can this filtering be done in the request rather than post processing?